Tasks/VsTest/vstest.ts

```diff
@@ -374,141 +226,130 @@ function executeVstest(testResultsDirectory: string, parallelRunSettingsFile: st
         // Keeping this code in case we want to change failOnStdErr
         errStream: new outStream.StringErrorWritable({ decodeStrings: false })
     };
-    vstest.exec(execOptions)
-        .then(function (code) {
-            cleanUp(parallelRunSettingsFile);
-            if (ignoreTestFailures === true) {
-                defer.resolve(tl.TaskResult.Succeeded); // ignore failures.
-            } else {
-                defer.resolve(code);
-            }
-        })
-        .fail(function (err) {
-            cleanUp(parallelRunSettingsFile);
-            tl.warning(tl.loc('VstestFailed'));
-            if (ignoreTestFailures) {
-                tl.warning(err);
-                defer.resolve(tl.TaskResult.Succeeded);
-            } else {
-                utils.Helper.publishEventToCi(AreaCodes.EXECUTEVSTEST, err.message, 1005, true);
-                tl.error(err);
-                defer.resolve(tl.TaskResult.Failed);
-            }
-        });
-    return defer.promise;
+    return vstest.exec(execOptions).then(function (code) {
+        cleanUp(parallelRunSettingsFile);
+        if (ignoreTestFailures === true) {
+            return tl.TaskResult.Succeeded; // ignore failures.
+        } else {
+            return code;
+        }
+    }).catch(function (err) {
+        cleanUp(parallelRunSettingsFile);
```

Comment:
Not sure whether a ci event can be added here. It is a failure but its also ignored.

Reply:
Discussed offline